R/check_inputs.R

```diff
@@ -14,4 +14,15 @@ check_inputs <- function(model, data, parameters, omega, regimen, censoring, typ
   if(!("function" %in% class(model))) {
     stop("The 'model' argument requires a function, e.g. a model defined using the new_ode_model() function from the PKPDsim package.")
   }
-}
\ No newline at end of file
+  check_parameters_matching(model, parameters)
+}
+
+check_parameters_matching <- function(model, parameters) {
+  defined_parameters <- names(attr(model, "parameters"))
+  if(! all(defined_parameters %in% names(parameters))) {
+    stop("One or more required parameters for the model have not been specified.")
+  }
+  if(any(names(parameters) %in% defined_parameters)) {
```

Comment:
Is this missing a `!`?
```suggestion
  if(any(!(names(parameters) %in% defined_parameters))) {
```

Reply:
It might also be nice to identify which names are not supported so it's easy to correct them as a user

Reply:
yeah, ! dropped out by accident. Will make it more verbose